lapce-proxy/src/lsp.rs

```diff
@@ -84,15 +85,51 @@ impl LspCatalog {
         language_id: &str,
         options: Option<Value>,
     ) {
+        let binary_args = self.get_plugin_binary_args(options.clone());
         let client = LspClient::new(
             language_id.to_string(),
             exec_path,
             options,
+            binary_args.to_owned(),
             self.dispatcher.clone().unwrap(),
         );
         self.clients.insert(language_id.to_string(), client);
     }
 
+    fn get_plugin_binary_args(&mut self, option: Option<Value>) -> Vec<String> {
+        let mut vals = Vec::new();
+
+        let option = match option {
+            Some(val) => val,
+            None => {
+                return vals;
+            }
+        };
+
+        let binary = match option["binary"].as_object() {
+            Some(binary) => binary,
+            None => return vals,
+        };
+
+        let option = match binary.get("binary_args") {
+            Some(binary_args) => binary_args,
+            None => return vals,
+        };
+
+        let option = if let Some(option) = option.as_array() {
+            option
+        } else {
+            println!("binary_args value should be of type [String].");
+            return vals;
+        };
+
+        for val in option {
+            vals.push(String::from(val.as_str().unwrap()));
+        }
+
+        return vals;
```

Comment:
- You don't need to spell out the `return`
 - I think with the above comment you can avoid the `vals` variable completely
 - If this function returned `Option<Vec...` instead, you might simplify the early returns into `?`s. Also this would help differentiating between "no args" and the error cases.